internal/blockchain/common/common.go

```diff
@@ -332,9 +333,20 @@ func buildBatchPin(ctx context.Context, event *blockchain.Event, params *BatchPi
 }
 
 func GetNamespaceFromSubName(subName string) string {
-	var parts = strings.Split(subName, "-")
+	// Strip the listener ID from the end of the name
+	const UUIDLength = 36
+	var parts []string
+	if len(subName) > UUIDLength {
+		uuidStart := len(subName) - UUIDLength
+		prefix := subName[:uuidStart-1]
+		listenerID := subName[uuidStart:]
+		if _, err := uuid.Parse(listenerID); err == nil {
+			parts = strings.Split(prefix, "-")
+		}
+	}
```

Comment:
Namespace names are also allowed to have dashes which I think would cause problems here